tftrt/examples-cpp/benchmark_runner/main.cpp

```diff
@@ -0,0 +1,203 @@
+    *output_names = getNodeNames(signature.outputs());
+
+    return tensorflow::Status::OK();
+}
+
+// Configure a `CallableHandle` that feeds from and fetches to a device.
+Status setupCallable(std::unique_ptr<tensorflow::Session>& session,
+                     std::vector<string>& input_names,
+                     std::vector<string>& output_names,
+                     string& device_name,
+                     tensorflow::Session::CallableHandle* handle) {
+    tensorflow::CallableOptions opts;
+    for (const auto& name : input_names) {
+        opts.add_feed(name);
+        opts.mutable_feed_devices()->insert({name, device_name});
+    }
+    for (const auto& name : output_names) {
+        opts.add_fetch(name);
+        opts.mutable_fetch_devices()->insert({name, device_name});
+    }
+    opts.set_fetch_skip_sync(true);
+    return session->MakeCallable(opts, handle);
+}
+
+int main(int argc, char* argv[]) {
+    // Parse arguments
+    string model_path = "/path/to/model/";
+    int32_t batch_size = 64;
+    int32_t image_size = 224;
+    int32_t image_channels = 3;
+    int32_t warmup_iters = 50;
+    int32_t eval_iters = 1000;
+    std::vector<Flag> flag_list = {
+        Flag("model_path", &model_path, "graph to be executed"),
+        Flag("batch_size", &batch_size, "batch size to use for inference"),
+        Flag("image_size", &image_size, "size of the input image"),
+        Flag("image_channels", &image_channels, "number of channels for the input image"),
+        Flag("warmup_iters", &warmup_iters, "number of warmup iterations to run"),
+        Flag("eval_iters", &eval_iters, "number of timed iterations to run"),
+    };
+    string usage = tensorflow::Flags::Usage(argv[0], flag_list);
+    const bool parse_result = tensorflow::Flags::Parse(&argc, argv, flag_list);
+    if (!parse_result) {
+        LOG(ERROR) << usage;
+        return -1;
+    }
+
+    // We need to call this to set up global state for TensorFlow.
+    tensorflow::port::InitMain(argv[0], &argc, &argv);
+    if (argc > 1) {
+        LOG(ERROR) << "Unknown argument " << argv[1] << "\n" << usage;
+        return -1;
+    }
+
+    // Setup session
+    tensorflow::SavedModelBundle bundle;
+    std::vector<string> input_names;
+    std::vector<string> output_names;
+    TFTRT_ENSURE_OK(loadModel(model_path, &bundle, &input_names, &output_names));
+
+    // Create inputs and move to device
+    Tensor input(tensorflow::DT_FLOAT, {batch_size, image_size, image_size, image_channels});
+    std::vector<Tensor> inputs_host = {input};
+    std::vector<Tensor> inputs_device;
+    std::vector<Tensor> outputs;
+    string device_name = getDeviceName(bundle.session);
+    TFTRT_ENSURE_OK(moveToDevice(device_name, inputs_host, &inputs_device));
+
+    // Configure to feed and fetch from device
+    tensorflow::Session::CallableHandle handle;
+    TFTRT_ENSURE_OK(setupCallable(bundle.session, input_names, output_names, device_name, &handle));

```

Comment:
@nvkevihu , what is the purpose of making this a callable as opposed to running it directly using session.Run()?

Reply:
I wanted the callable to feed input tensors from device (by configuring `CallableOptions`), so that timing would not include copying memory from host to device. As far as I know, this is not possible with `session.Run()` and `RunOptions`.